website/docs/developer/research/research-reconciliation.mdx

```diff
@@ -0,0 +1,683 @@
+offerings.
+
+<MyImage
+    imageSource={useBaseUrl('img/Research-Reconciliation-Cumulus-Reconcile.png')}
+    imageAlt="System Context"
+    zoomInPic={useBaseUrl('img/zoom-in.svg')}
+    zoomOutPic={useBaseUrl('img/zoom-out.svg')}
+    resetPic={useBaseUrl('img/zoom-pan-reset.svg')}
+/>
+
+
+## Populating the metadata
+
+Populating the metadata objects with the proper metadata will need to come from
+various sources. The sections below will go through thoughts and ideas of where
+the metadata will originate from to populate the objects and potential ideas on
+what the underlying architecture may look like to perform the tasks.
+
+### Automated Cumulus Ingest Workflow
+
+The conventional way of populating the objects with metadata would be through
+the normal Cumulus ingest workflow that contains ORCA's `copy_to_glacier` lambda.
+The `copy_to_glacier` lambda would need to be modified to take or pull additional
+metadata needed for a granule's files like `providerId` and `collectionId` and
+various file information and send it to a queue. The body sent to the queue
+would have enough information for another ORCA lambda to write and/or update the
+metadata records for the various objects.
+
+Note that this new database write lambda or the `copy_to_glacier` lambda may need
+to make some additional queries to the S3 ORCA bucket to pull information like
+version and other values. Looking at the original `copy_to_glacier`, `s3.copy`
+is used to copy data to the ORCA archive. Based on initial research located
+[here](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.copy),
+the `s3.copy` command does not return any information. See the test performed
+below. Because of this, we will likely need to perform a listing on the object to
+get additional information like the version as seen in the sample seen below.
+Some things to consider when looking at the options
+for getting the additional data include speed (how fast can I get the data), cost
+(what is the cost impact of additional gets on the  s3 bucket), and complexity
+(where is the best place to put these additional gets, is it overly complex)
+
+#### Boto3 Copy Test
+
+```python
+import boto3
+
+# Create the Client
+s3_client = boto3.resource('s3')
+
+# Get a source and destination to copy
+copy_source = {"Bucket": "orca-sandbox-s3-provider", "Key": "MOD09GQ/006/MOD09GQ.A2017025.h21v00.006.2017034065111.hdf"}
+copy_destination = "orca-sandbox-glacier"
+
+# perform the copy command
+copy_command = s3_client.meta.client.copy(copy_source, copy_destination, copy_source["Key"])
+
+# Return results
+print(copy_command)
+None
+```
+
+
+#### Boto3 Get Additional File Information
+
+This would occur after an `s3.copy` and assumes that the copy is synchronous and
+that the file is not be ingested in parallel in a separate ingest pipeline that
+would lead to a collision.
+
+```python
+import boto3
+
+# Create the Client
+s3_client = boto3.resource('s3')
```

Comment:
should this be boto3.client('s3') since it's saying create the client?

Reply:
Either could be used. See the actual call that occurs later `s3_client.meta.client.` This is really just sample code to show the basics of the call and the outputs to provide context for further design.